### ISSUE TITLE
Authorization code not linked to user_id

### ISSUE THREAD

**Issue by EdwinOtten** [chadicus/slim-oauth2-routes#28]
### Expected behavior
When a user authorizes a client, the user_id should be saved with the authorization code if the user_id argument is set.

### Actual behavior
The authorization code is created but without any user_id (the column `user_id` in table `oauth_authorization_codes` remains NULL)

### Steps to reproduce the behavior
Invoke the Authorize route, with or without a user_id argument.

### Suggested fix
The Authorize route calls `handleAuthorizeRequest()` on the OAuth2 server, but doesn't pass the user_id argument (which then defaults to null). Therefore the OAuth2 server won't link the authorization code to the user_id.

I suggest adding the user_id at **line 84** of **`Authorize.php`**:
```
$user_id = isset($arguments['user_id']) ? $arguments['user_id'] : null;
$this->server->handleAuthorizeRequest($oauth2Request, $oauth2Response, $authorized === 'yes', $user_id);
```

### Context
In my application I use Slim middleware to validate the user credentials and then set the user_id argument:
```
$loginMiddelware = function ($request, $response, $next) 
{
  $email = $request->getParam('email');
  $password = $request->getParam('password');

  // Validate the user credentials
  $user_id = getUserIdIfValidCredentials($email, $password);
  if ($user_id === FALSE || !is_numeric($user_id)) {
    return $response->withStatus(303)->withHeader('Location', URL_AUTHORIZE_INVALID_CREDENTIALS);
  }

  // Set the user_id argument so that Routes\Authorize can give it to handleAuthorizeRequest
  $route = $request->getAttribute('route');
  $route->setArgument('user_id', $user_id);

  // Credentials are valid, continue so the authorization code can be sent to the clients callback_uri
  $response = $next($request, $response);
  return $response;
};
```

**Comment by chadicus** [comment]
I will try to get this taken care of over the weekend

**Comment by chadicus** [comment]
@EdwinOtten  would the solution found in #29 work for you?  You would have to build your own UserIdProvider

**Comment by EdwinOtten** [comment]
@chadicus Yes that would work, better than my example because it's a more OO approach. Looks good! 

Also, how would you implement the user credentials validation? As you can see I am using middleware for this. Would you do that similarly or do you also have an Interface for validating the user login?

**Comment by chadicus** [comment]
@EdwinOtten I think your middleware approach would work just fine.

**Comment by chadicus** [comment]
Released with [v3.1.1](https://github.com/chadicus/slim-oauth2-routes/releases/tag/v3.1.1)
